inc/functions.php

```diff
@@ -5033,6 +5017,48 @@ function build_theme_select($name, $selected="", $tid=0, $depth="", $usergroup_o
 }
 
 /**
+ * Get the theme data of a theme id.
+ *
+ * @param int The theme id of the theme.
+ * @return array The theme data
+ */
+function get_theme($tid)
+{
+	global $tcache, $db;
+
+	if(!is_array($tcache))
+	{
+		$query = $db->simple_select('themes', 'tid, name, pid, allowedgroups', "pid!='0'", array('order_by' => 'pid, name'));
+
+		while($theme = $db->fetch_array($query))
+		{
+			$tcache[$theme['pid']][$theme['tid']] = $theme;
+		}
+	}
+
+	$s_theme = false;
+
+	foreach($tcache as $themes)
+	{
+		foreach($themes as $theme)
+		{
+			if($tid == $theme['tid'])
+			{
+				$s_theme = $theme;
+				break;
```

Comment:
`break 2;` could be used here to break both foreach structures instead of the additional `if` below.